provider/lxd/credentials.go

```diff
@@ -218,7 +220,34 @@ func (p environProviderCredentials) FinalizeCredential(
 	args environs.FinalizeCredentialParams,
 ) (*cloud.Credential, error) {
 	switch authType := args.Credential.AuthType(); authType {
-	case interactiveAuthType, cloud.CertificateAuthType:
+	case interactiveAuthType:
+		credAttrs := args.Credential.Attributes()
+		// We don't care if the password is empty, just that it exists. Empty
+		// passwords can be valid ones...
+		if _, ok := credAttrs[credAttrTrustPassword]; ok {
+			// check to see if the client cert, keys exist, if they do not,
+			// generate them for the user.
+			if _, ok := getClientCertificates(args.Credential); !ok {
+				stderr := ctx.GetStderr()
+				nopLogf := func(s string, args ...interface{}) {
+					fmt.Fprintf(stderr, s+"\n", args...)
+				}
+				clientCert, clientKey, err := p.readOrGenerateCert(nopLogf)
```

Comment:
nopLogf?

Reply:
This was in the old code, I'm just reusing it in this instance, guess the original person didn't want logging in every scenario.

Reply:
Personally I don't see why we do that, as it's valuable information. Any opposition in removing it?